app/executor.go

```diff
@@ -0,0 +1,21 @@
+package app
+
+import (
+	"context"
+
+	storetypes "cosmossdk.io/store/types"
+	abci "github.com/cometbft/cometbft/abci/types"
+	evmtypes "github.com/evmos/ethermint/x/evm/types"
+)
+
+func DefaultTxExecutor(_ context.Context,
+	blockSize int,
+	ms storetypes.MultiStore,
+	deliverTxWithMultiStore func(int, storetypes.MultiStore) *abci.ExecTxResult,
+) ([]*abci.ExecTxResult, error) {
+	results := make([]*abci.ExecTxResult, blockSize)
+	for i := 0; i < blockSize; i++ {
+		results[i] = deliverTxWithMultiStore(i, ms)
+	}
+	return evmtypes.PatchTxResponses(results), nil
```

Comment:
it'd be better if we can patch the result before they are encoded, will leave that to later if ever do that.